src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Linux.cs

```diff
@@ -39,26 +37,29 @@ public class NetworkChange
             {
                 lock (s_gate)
                 {
-                    if (s_socket == 0)
+                    if (s_socket == 0 && value != null)
                     {
                         CreateSocket();
                     }
 
-                    s_addressChangedSubscribers += value;
+                    if (value != null)
+                    {
+                        s_addressChangedSubscribers.TryAdd(value, ExecutionContext.Capture());
+                    }
                 }
             }
             remove
             {
                 lock (s_gate)
                 {
```

Comment:
Do we need to do any of this if value is null?  Wondering if there should be parity with add.